.github/workflows/main.yml

```diff
@@ -57,37 +58,57 @@ jobs:
         else
           echo "::set-output name=result::odd"
         fi
+  build-and-cache-node-modules:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Setup repo
+        uses: actions/checkout@v3
+         
+      - uses: actions/setup-node@v3
+        with: 
+          registry-url: ${{ env.NPM_REGISTRY }}
+          scope: "${{ env.SCOPE }}"
+          cache: 'npm'
+
+      - name: Setup Deno
+        uses: denoland/setup-deno@v1
+        with:
+          deno-version: "1.23.4"
+
+      - name: Build all
+        run: |
+          export PATH="$PATH:$(pwd)"
+          npm run build-all
```

Comment:
Why are you building the packages here if what you're caching are only the `node_modules` ? Shouldn't just an `npm ci` work here?

Reply:
This is the initial build that will build all the `node_modules` that will be used later on. Because the cache is unique for each run I need to build all the modules at least once.